src/Game/UI/uiFadeProgress.cpp

```diff
@@ -0,0 +1,55 @@
+#include "Game/UI/uiFadeProgress.h"
+#include "Game/UI/uiUtils.h"
+
+namespace uking::ui {
+
+SEAD_SINGLETON_DISPOSER_IMPL(FadeProgress)
+
+void FadeProgress::init() {
+    mActive = false;
+    mProgress = 0.0;
+    mGaugeValue = 0.0;
+    mSteps = 1.0;
+}
+
+void FadeProgress::reset() {
+    init();
+}
+
+void FadeProgress::updateFadeScreen() {
+    if (mActive) {
+        f32 progress = mProgress;
+        f32 difference = progress - mGaugeValue;
+
+        if (difference > 0.1f)
+            mSteps = 1.1;
+        if (difference > 0.2f)
+            mSteps = 1.2;
+        if (progress > 0.94f)
+            mSteps = 10.0;
+
+        if (difference <= 0.0001f && difference >= -0.0001f)
+            mSteps = 1.0;
+
+        f32 newProgress = mGaugeValue + (mProgressPerStep * mSteps);
```

Comment:
`new_progress` (Nintendo uses snake_case for local variables. yes, this is despite member variables being mWrittenLikeThis)